proxy/src/main/java/com/wavefront/agent/PushAgent.java

```diff
@@ -691,6 +690,23 @@ protected void processConfiguration(AgentConfiguration config) {
         logger.fine("Proxy push batch set to (locally) " + pushFlushMaxPoints.get());
       }
 
+      if (config.getCollectorSetsRateLimit() != null && config.getCollectorSetsRateLimit()) {
+        Long collectorRateLimit = config.getCollectorRateLimit();
+        if (pushRateLimiter != null && collectorRateLimit != null && pushRateLimiter.getRate() != collectorRateLimit) {
+          pushRateLimiter.setRate(collectorRateLimit);
+          logger.info("Proxy rate limit set to " + collectorRateLimit + " remotely");
+        }
+      } else {
+        if (pushRateLimiter != null && pushRateLimiter.getRate() != pushRateLimit) {
+          pushRateLimiter.setRate(pushRateLimit);
+          if (pushRateLimit >= 10_000_000) {
+            logger.info("Proxy rate limit no longer enforced");
```

Comment:
warning? (feel free to ignore my suggestion)

Reply:
agreed - rate limit changes are more important than regular info messaged. done.